### DISCUSSION TITLE
Support Request for - Error: 13 INTERNAL: the server has encountered an internal error

### DISCUSSION THREAD

**Discussion by Prithvi-Shah** [speechly/speechly#199] [Q&A] [answered]
Hi Speechly Team, 

Hope all is well. 

I'm starting this discussion to get some support around using Speechly's gRPC API. 
This is what I'm trying to do - 

Use an existing token that I copied from the UI - https://api.speechly.com/dashboard/#/dashboard to test a nodejs server side implementation here - https://www.npmjs.com/package/@speechly/api/v/0.8.0

I am trying to perform the SLU on this wav file - https://docs.speechly.com/test1_en.wav

PFA the code for the same. 
I've masked the app ID & Token. 

I keep getting an Internal Server error. Thought it might have something to do with the token but this is my error log - 
**Started audio context b6e50fb2-64bd-42e5-8fe3-b80e0342be3f
Stopped audio context b6e50fb2-64bd-42e5-8fe3-b80e0342be3f
Error: 13 INTERNAL: the server has encountered an internal error
    at Object.callErrorFromStatus (D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\call.js:31:26)
    at Object.onReceiveStatus (D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\client.js:409:49)
    at Object.onReceiveStatus (D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\client-interceptors.js:328:181)
    at D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\call-stream.js:187:78
    at processTicksAndRejections (internal/process/task_queues.js:75:11) {
  code: 13,
  details: 'the server has encountered an internal error',
  metadata: Metadata { internalRepr: Map(0) {}, options: {} }
}
[Finished in 4.581s]**

Code - Speechly_API_Test.js

Doesn't look like the token failed. Anyway, I tried to also login and get a newer token but that seems to fail. Is there any specification for setting the device ID? 
Also, attaching a separate script for Login issues to get the token basically. 
Here's the error log for the same - 

D:\Working Folder\Speech Science\STT\Speechly\Login.js:28
        token: res.getToken(),
                   ^

TypeError: Cannot read property 'getToken' of undefined
    at Object.callback (D:\Working Folder\Speech Science\STT\Speechly\Login.js:28:20)
    at Object.onReceiveStatus (D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\client.js:189:36)
    at Object.onReceiveStatus (D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\client-interceptors.js:365:141)
    at Object.onReceiveStatus (D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\client-interceptors.js:328:181)
    at D:\Working Folder\Speech Science\STT\Speechly\node_modules\@grpc\grpc-js\build\src\call-stream.js:187:78
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
[Finished in 2.374s]

Code- Speechly_Login_Test.js
[Speechly Support Request.zip](https://github.com/speechly/speechly/files/8829734/Speechly.Support.Request.zip)

I tried figuring it out with these discussions but doesn't seem like anyone else had any 'server encountered an Internal error' issues. 
I'm not sure what I'm doing incorrectly. This is my first time working with gRPC so I appreciate any help to resolve this. 

Mahalo for your support. 

Regards, 
Prithvi

**Comment by langma** [comment, marked as answer]
Hi Prithvi!

I've checked what happened in the backend, and the reason for the error is not in your code. The error comes from the fact that your appId has no model trained - you should deploy a configuration in the dashboard and wait for it to be deployed. 

At the moment the easiest way to achieve this is to edit the configuration in the dashboard (just remove or add comment) and the "deploy" button will be enabled. Click that and wait for it to be ready, and after that your code should work better :)

Thanks for reporting this, we need to handle this error situation better!

**Reply by Prithvi-Shah** [reply]
Hey Langma,

Thank you for the response!
I was able to get the code working after I deployed the configuration :)

Just a follow up question which is a bit of a segue - 
*Can I use the free account to get SLU for 8 kHz audio? According to this - https://github.com/speechly/speechly/discussions/192, I cannot do so because frequency bands are lost when audio is stored as 8 kHz. 
Is the enterprise tier the only alternative and if so, the only change in the code would be a change in the sample rate configuration, correct? 

Regards,
Prithvi

**Reply by langma** [reply]
Yes, unfortunately 8KHz is not trivial to support and that is why we consider that to be in the enterprise offering at the moment. 

The only change to the calling code is, as you suspected, only in the sample rate configuration.
